chainer/functions/normalization/batch_normalization.py

```diff
@@ -15,6 +15,233 @@
 import chainerx
 
 
+class _BatchNormalizationBackend:
+
+    def forward(self, axis, gamma, x, xp, expander,
+                beta, eps, decay, running_mean, running_var):
+        raise NotImplementedError()
+
+    def backward(self, axis, gamma, gy, x, xp,
+                 expander, mean, inv_std, eps, var):
+        raise NotImplementedError()
+
+
+class GeneralBatchNormalizationBackend(_BatchNormalizationBackend):
+
+    def __init__(self):
+        self.mean = None
+        self.inv_std = None
+        self.var = None
+        self.running_mean = None
+        self.running_var = None
+
+    def forward(self, axis, gamma, x, xp, expander,
+                beta, eps, decay, running_mean, running_var):
+        self.running_mean = running_mean
```

Comment:
Rather than interfacing using attributes in `self`, I think it's simpler to include all the input/output requirements in arguments and return values.
So, how about returning them (`mean`, `inv_std`, `var`, `running_mean`, `running_var`) along with `y` instead of assigning attributes?

Reply:
addressed in 84628a4